packages/@react-spectrum/searchfield/src/MobileSearchAutocomplete.tsx

```diff
@@ -0,0 +1,549 @@
+  }, popoverRef);
+
+  // Override the role of the input to "searchbox" instead of "combobox".
+  // Since the listbox is always visible, the combobox role doesn't really give us anything.
+  // VoiceOver on iOS reads "double tap to collapse" when focused on the input rather than
+  // "double tap to edit text", as with a textbox or searchbox. We'd like double tapping to
+  // open the virtual keyboard rather than closing the tray.
+  inputProps.role = 'searchbox';
+  inputProps['aria-haspopup'] = 'listbox';
+  delete inputProps.onTouchEnd;
+
+  let clearButton = (
+    <ClearButton
+      {...clearButtonProps}
+      preventFocus
+      aria-label={formatMessage('clear')}
+      excludeFromTabOrder
+      UNSAFE_className={
+        classNames(
+          searchStyles,
+          'spectrum-ClearButton'
+        )
+      }
+      isDisabled={isDisabled} />
+  );
+
+  let loadingCircle = (
+    <ProgressCircle
+      aria-label={formatMessage('loading')}
+      size="S"
+      isIndeterminate
+      UNSAFE_className={classNames(
+        searchStyles,
+        'spectrum-Search-circleLoader',
+        classNames(
+          textfieldStyles,
+          'spectrum-Textfield-circleLoader'
+        )
+      )} />
+  );
+
+  // Close the software keyboard on scroll to give the user a bigger area to scroll.
+  // But only do this if scrolling with touch, otherwise it can cause issues with touch
+  // screen readers.
+  let isTouchDown = useRef(false);
+  let onTouchStart = () => {
+    isTouchDown.current = true;
+  };
+
+  let onTouchEnd = () => {
+    isTouchDown.current = false;
+  };
+
+  let onScroll = useCallback(() => {
+    if (!inputRef.current || document.activeElement !== inputRef.current || !isTouchDown.current) {
+      return;
+    }
+
+    popoverRef.current.focus();
+  }, [inputRef, popoverRef, isTouchDown]);
+
+  let inputValue = inputProps.value;
+  let lastInputValue = useRef(inputValue);
+  useEffect(() => {
+    if (loadingState === 'filtering' && !showLoading) {
+      if (timeout.current === null) {
+        timeout.current = setTimeout(() => {
+          setShowLoading(true);
+        }, 500);
+      }
+
+      // If user is typing, clear the timer and restart since it is a new request
+      if (inputValue !== lastInputValue.current) {
+        clearTimeout(timeout.current);
+        timeout.current = setTimeout(() => {
+          setShowLoading(true);
+        }, 500);
+      }
+    } else if (loadingState !== 'filtering') {
+      // If loading is no longer happening, clear any timers and hide the loading circle
+      setShowLoading(false);
+      clearTimeout(timeout.current);
+      timeout.current = null;
+    }
+
+    lastInputValue.current = inputValue;
+  }, [loadingState, inputValue, showLoading]);
+
+  let onKeyDown = (e) => {
+    // Close virtual keyboard if user hits Enter w/o any focused options
+    if (e.key === 'Enter' && state.selectionManager.focusedKey == null) {
+      popoverRef.current.focus();
```

Comment:
Might need to fire `onSubmit` manually here as well

Reply:
Just added, thanks!